user_profile: Add keyboard navigation for remove buttons.

Hell1213 opened the pull request:
Adds keyboard navigation for remove buttons in channels and groups tables in the user profile modal.

Fixes: #36641

**How changes were tested:**

- Tested Tab/Shift-Tab navigation to remove buttons
- Tested arrow key navigation (ArrowRight/ArrowLeft) within rows
- Verified buttons appear when focused
- Tested Enter key triggers unsubscribe/remove
- Confirmed mouse hover and click still work
- Ran linters and frontend tests

**Screenshots :**
Before:
<img width="1002" height="952" alt="Screenshot from 2025-11-24 22-52-18" src="https://github.com/user-attachments/assets/cb024f0f-863f-4b13-954c-4e74741e4be6" />
<img width="1002" height="952" alt="Screenshot from 2025-11-24 22-52-43" src="https://github.com/user-attachments/assets/3c391a99-4d55-4c95-8bce-97efbd5e8ab3" />

After:
<img width="1002" height="952" alt="Screenshot from 2025-11-24 22-48-25" src="https://github.com/user-attachments/assets/48c71bd3-bea4-4184-9046-5a401819571a" />
<img width="1002" height="952" alt="Screenshot from 2025-11-24 22-48-31" src="https://github.com/user-attachments/assets/0be75d4e-f08e-4509-b0fc-0b74b06cdda6" />
<img width="1002" height="952" alt="Screenshot from 2025-11-24 22-49-14" src="https://github.com/user-attachments/assets/cbd0cda6-19b0-44a1-95d0-927b2e0dfce9" />
<img width="1002" height="952" alt="Screenshot from 2025-11-24 22-49-18" src="https://github.com/user-attachments/assets/a3f1155f-33ac-4a3d-9704-097858f9c7f5" />



**screen capture: ** 
Before:

https://github.com/user-attachments/assets/79cda75a-11de-40c7-aa01-759f2a67226c

After:


https://github.com/user-attachments/assets/aa1b4f3b-d483-4e41-9517-6dc82f3b8c84



<details>
<summary>Self-review checklist</summary>

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

Hell1213 committed:
user_profile: Fix keyboard navigation for remove buttons.

Simplify arrow key handling and remove custom Enter key handling.
Move channel folder keyboard code to channel_folders_ui.ts.
Fix remove button focus behavior in user profiles and channel folders.

Fixes #36641.

alya commented:
@pratik-pc Are you up for reviewing this one? I haven't tested.

pratik-pc commented:
@Hell1213 This currently doesn't implement the arrow key navigation. Can you also add that according to the issue description?

Hell1213 commented:
> @Hell1213 This currently doesn't implement the arrow key navigation. Can you also add that according to the issue description?

Thanks @pratik-pc — got it.  
I’ll add the arrow key navigation as described in the issue and update the PR soon.

pratik-pc commented:
> @pratik-pc
> Update pushed — arrow key navigation is now implemented.

I am still not able to navigate using the arrow keys

Hell1213 commented:
> Screencast.from.2025-12-09.19-14-57.webm 
> I think Shift+Tab should move focus to remove button to the row above. Can you also rebase it onto `main`, some of the class names have since been changed. Also, this PR fixes only part of #36641
  
updated the pr accordingly as requested shift+tab move to focus to remove button of the row above and now pr fixing full #36641.
Added new demo video in description.

pratik-pc commented on web/src/user_profile.ts:
```diff
@@ -1469,6 +1474,126 @@ export function initialize(): void {
+            e.stopPropagation();
+
+            const $elt = $(e.currentTarget);
+            $elt.trigger("click");
```
We can either let the browser handle the Enter key here, or, perhaps better, focus the next equivalent available element.

pratik-pc reviewed (commented):
Left some comments

Hell1213 commented:
hey @pratik-pc , its ready pls check when you get time , ready for any further requests . 
thanks .
